src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Rectangle implements Figure {
+  shape: string = 'rectangle';
+
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Invalid width or height');
+    }
+    this.color = color;
+    this.width = width;
+    this.height = height;
+  }
 
+  getArea(): number {
+    return Math.floor(this.width * this.height * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: string = 'circle';
+
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid radius');
+    }
+    this.color = color;
+    this.radius = radius;
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
```

Comment:
Instead of using 'sideA', 'sideB', and 'sideC', use 'this.sideA', 'this.sideB', and 'this.sideC' to refer to the class properties.